src/test/java/org/ambraproject/rhino/service/impl/HibernatePersistenceServiceTest.java

```diff
@@ -0,0 +1,387 @@
+
+  private IngestPackage expectedIngestPackage;
+
+  private Optional<Journal> expectedJournal;
+
+  /**
+   * Creates an instance of <code>HibernatePersistenceServiceTest</code>.
+   */
+  public HibernatePersistenceServiceTest() {
+    super(true /* spyOnHibernateTemplate */);
+  }
+
+  /**
+   * Initialize test data fixtures.
+   */
+  @BeforeMethod(alwaysRun = true)
+  public void init() {
+    articleDoi = Doi.create(ARTICLE_DOI_URI);
+
+    final LocalDate publishedOn = LocalDate.now();
+
+    expectedArticle = new Article();
+    expectedArticle.setDoi(articleDoi.getName());
+
+    mockArchive = IngestionServiceTest.createStubArchive(new byte[] {} /* manifestXml */,
+        IngestionServiceTest.ARTICLE_INGEST_ENTRIES);
+
+    mockArticleXml = mock(ArticleXml.class);
+    mockManifestFile = mock(ManifestXml.ManifestFile.class);
+    mockManifestAsset = mock(ManifestXml.Asset.class);
+    mockAncillaryFile = mock(ManifestXml.ManifestFile.class);
+    mockRepresentation = mock(ManifestXml.Representation.class);
+    mockManuscriptAsset = mock(ManifestXml.Asset.class);
+    mockManifest = mock(ManifestXml.class);
+
+    when(mockArticleXml.readDoi()).thenReturn(articleDoi);
+
+    when(mockManifestFile.getCrepoKey()).thenReturn("10.1111/dupp.0000001.pdf");
+    when(mockManifestFile.getEntry()).thenReturn("dupp.0000001.pdf");
+    when(mockManifestFile.getMimetype()).thenReturn("application/pdf");
+
+    when(mockManifestAsset.getAssetTagName()).thenReturn(AssetTagName.OBJECT);
+    when(mockManifestAsset.getUri()).thenReturn("info:doi/10.1111/dupp.0000001");
+    when(mockManifestAsset.isStrikingImage()).thenReturn(Boolean.TRUE);
+
+    when(mockAncillaryFile.getCrepoKey()).thenReturn("10.1111/dupp.0000001.s001.png");
+    when(mockAncillaryFile.getEntry()).thenReturn("dupp.0000001.s001.png");
+    when(mockAncillaryFile.getMimetype()).thenReturn("image/ext");
+
+    when(mockRepresentation.getFile()).thenReturn(mockManifestFile);
+
+    when(mockManuscriptAsset.getRepresentation("manuscript"))
+        .thenReturn(Optional.of(mockRepresentation));
+    when(mockManuscriptAsset.getRepresentation("printable")).thenReturn(Optional.empty());
+    when(mockManuscriptAsset.getUri()).thenReturn(ARTICLE_DOI_URI);
+
+    when(mockManifest.getArticleAsset()).thenReturn(mockManuscriptAsset);
+    when(mockManifest.getAssets()).thenReturn(ImmutableList.of());
+    when(mockManifest.getAncillaryFiles()).thenReturn(ImmutableList.of(mockAncillaryFile));
+
+    expectedArticlePackage =
+        new ArticlePackageBuilder(DESTINATION_BUCKET, mockArchive, mockArticleXml, mockManifest)
+            .build();
+
+    expectedArticleMetadata = ArticleMetadata.builder().setTitle("Meta title")
+        .setArticleType("MetaArticleType")
+        .setEissn(EISSN)
+        .setJournalName(META_JOURNAL_NAME)
+        .setPublicationDate(publishedOn).build();
+
+    expectedCustomMetadata = ArticleCustomMetadata.builder().setPublicationStage("pubtage")
```

Comment:
should we say "pubtage"-->"pubstage" . Ignore it if its too much trouble.

Reply:
Done.